contractcourt/channel_arbitrator_test.go

```diff
@@ -2686,6 +2686,112 @@ func TestChannelArbitratorAnchors(t *testing.T) {
 	)
 }
 
+// TestChannelArbitratorStartAfterCommitmentRejected tests that
+// when we run into the case where our commitment tx is rejected by our bitcoin
+// backend we still continue to startup the arbitrator for a specific set of
+// errors.
+func TestChannelArbitratorStartAfterCommitmentRejected(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+
+		// The specific error during broadcasting the transaction.
+		broadcastErr error
+
+		// expectedd state when the startup of the arbitrator succeeds.
+		expectedStates ArbitratorState
+
+		expectedStartup bool
+	}{
+		{
+			name: "Commitment is rejected because of low mempool " +
+				"fees",
+			broadcastErr:    lnwallet.ErrMempoolFee,
+			expectedStates:  StateCommitmentBroadcasted,
+			expectedStartup: true,
+		},
+		{
+			// We map a rejected rbf transaction to ErrDoubleSpend
+			// in lnd.
+			name: "Commitment is rejected because of a " +
+				"rbf transaction not succeeding",
+			broadcastErr:    lnwallet.ErrDoubleSpend,
+			expectedStates:  StateCommitmentBroadcasted,
+			expectedStartup: true,
+		},
+		{
+			name: "Commitment is rejected with an " +
+				"unmatched error",
+			broadcastErr:    fmt.Errorf("Reject Commitment Tx"),
+			expectedStates:  StateBroadcastCommit,
+			expectedStartup: false,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			// We'll create the arbitrator and its backing log
+			// to signal that it's already in the process of being
+			// force closed.
+			log := &mockArbitratorLog{
+				newStates: make(chan ArbitratorState, 1),
+				state:     StateBroadcastCommit,
+			}
+			chanArbCtx, err := createTestChannelArbitrator(t, log)
+			require.NoError(t, err, "unable to create "+
+				"ChannelArbitrator")
+
+			chanArb := chanArbCtx.chanArb
+
+			stateChan := make(chan ArbitratorState, 1)
+
+			// Customize the PublishTx function of the arbitrator.
+			chanArb.cfg.PublishTx = func(*wire.MsgTx,
+				string) error {
+
+				select {
```

Comment:
Do you have specific reason to add the check what the state was before publishing? We could capture the state (defined in the outer scope) in the closure, there's no need for synchronization, I think. Would it be better to leave out all `stateChan` related code and solely rely on `AssertStateTransition` as you have it?